### PR TITLE
Fixed realtime protocol version check

### PR DESCRIPTION
**Pull request by lordnull**
The wrong check was causing the ack to be proxied through the helper.
The helper builds a map for acks from v1 to the actual v2 sequence
number. However, because this was a version 2 protocol, the map was
not needed, and thus not built. Without finding the v1 sequence number,
and ack is never sent to the rtq, leaving the object forever unacked.

### CODE REVIEW

File: src/riak_repl2_rtsource_conn.erl [hunk 1]
```diff
@@ -340,10 +340,11 @@ recv(TcpBin, State = #state{remote = Name,
                             hb_timeout_tref = HBTRef}) ->
     %% hb_timeout_tref might be undefined if we have are getting
     %% acks/heartbeats back-to-back and we haven't sent a heartbeat yet.
+    {realtime, {ProtoMajor, _}, {ProtoMajor, _}} = State#state.proto,
     case riak_repl2_rtframe:decode(TcpBin) of
         {ok, undefined, Cont} ->
             {noreply, State#state{cont = Cont}};
-        {ok, {ack, Seq}, Cont} when State#state.proto == {2,0} ->
+        {ok, {ack, Seq}, Cont} when ProtoMajor == 2 ->
```
Comment (jonmeredith) [inline src/riak_repl2_rtsource_conn.erl]: which makes least surprise `==2` or `>= 2` ?

Comment (lordnull) [inline src/riak_repl2_rtsource_conn.erl]: Depends on what needs to happen if/when we get to major version 3 of the protocol. If it's a big enough change, there'll be a whole new clause for it. If it doesn't need to change much (in which case why is it a new major version) then >= 2 is better.

Comment (jonmeredith) [inline src/riak_repl2_rtsource_conn.erl]: I think >=2 is better, least surprise. It's on whoever introduces major=3 to make sure it's right.

**Comment by bowrocker** [comment]
Passes the rt_cascading test for it in this PR:

https://github.com/basho/riak_test/pull/386

+1

**Pull request merged**